src/sdk/modules/validators/smartSessionValidator/smartSessionValidator.test.ts

```diff
@@ -0,0 +1,324 @@
+
+    // Todo: Add a negative test case for time range policy
+    const sessionRequestedInfo: CreateSessionDataParams = {
+      sessionPublicKey: sessionKeyEOA,
+      sessionValidatorAddress: TEST_CONTRACTS.SimpleSessionValidator.address,
+      sessionKeyData: toHex(toBytes(sessionKeyEOA)),
+      sessionValidAfter: 0,
+      sessionValidUntil: 0,
+      actionPoliciesInfo: [
+        {
+          contractAddress: TEST_CONTRACTS.Counter.address, // counter address
+          functionSelector: "0x273ea3e3" as Hex, // function selector for increment count
+          validUntil: 0, // 1717001666
+          validAfter: 0,
+          rules: [], // no other rules and conditions applied
+          valueLimit: BigInt(0)
+        }
+      ]
+    }
+
+    const smartSessionNexusClient = nexusClient.extend(
+      smartSessionValidatorActions()
+    )
+
+    const enableSessionsResponse = await smartSessionNexusClient.enableSessions(
+      {
+        account: nexusClient.account,
+        sessionRequestedInfo: [sessionRequestedInfo]
+      }
+    )
+
+    expect(enableSessionsResponse.userOpHash).toBeDefined()
+    expect(enableSessionsResponse.permissionIds).toBeDefined()
+
+    const permissionIds = enableSessionsResponse.permissionIds
+    expect(permissionIds.length).toBe(1)
+    const permissionId = permissionIds[0]
+    cachedPermissionId = permissionId
+
+    const receipt = await nexusClient.waitForUserOperationReceipt({
+      hash: enableSessionsResponse.userOpHash
+    })
+
+    expect(receipt.success).toBe(true)
+
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: permissionId
+    })
+    expect(isEnabled).toBe(true)
+  }, 60000)
+
+  test("should make use of already enabled session (USE mode) to increment a counter using a session key", async () => {
+    const isEnabled = await isSessionEnabled({
+      client: nexusClient.account.client as PublicClient,
+      accountAddress: nexusClient.account.address,
+      permissionId: cachedPermissionId
+    })
+    expect(isEnabled).toBe(true)
+
+    const smartSessionValidator = await toSmartSessionValidatorModule({
+      client: nexusClient.account.client as PublicClient,
+      initData: encodePacked(["address"], [eoaAccount.address]),
+      deInitData: "0x",
+      nexusAccountAddress: nexusClient.account.address,
+      activePermissionId: cachedPermissionId
+    })
+
+    const pubClient = nexusClient.account.client as PublicClient
+
+    const counterBefore = await pubClient.readContract({
+      address: TEST_CONTRACTS.Counter.address,
+      abi: CounterAbi,
+      functionName: "getNumber",
+      args: []
+    })
+
+    // helpful for out of range test
+    // await testClient.setNextBlockTimestamp({
+    //   timestamp: 9727001666n
+    // })
+
+    // set active validation module
+    nexusClient.account.setActiveValidationModule(smartSessionValidator)
```

Comment:
Wait, users still need to do this? Can't this be removed?

Reply:
We should take this away from the user. Can we put it inside `useEnabledSession`? 

`smartSessionValidator` can be calculated using only the nexusClient, (which will be the `client` inside `useEnabledSession`).

This test should **begin** with the statement `smartSessionNexusClient.useEnabledSession(...)`

Reply:
I'm sure you're making lot of sense. let me go to the test case and confirm and make changes.

Reply:
in useSession I retrieve it from 

  const smartSessionValidator = (
    account as NexusAccount
  ).getActiveValidationModule() as ToSmartSessionValidatorModuleReturnType
  
  so it should have been set at some point. and instance of smartSessionValidator to be created before that sometime